src/plugins/workspace/public/plugin.ts

```diff
@@ -106,9 +139,28 @@ export class WorkspacesPlugin implements Plugin<{}, {}, WorkspacesPluginSetupDep
     }
   }
 
+  private getFilterLinks(
+    workspace: WorkspaceAttribute | null | undefined,
+    allNavLinks: ChromeNavLink[]
+  ) {
+    if (!workspace) return allNavLinks;
+    const features = workspace.features ?? [];
+    return allNavLinks.filter((item) => features.includes(item.id));
+  }
+
   public start(core: CoreStart) {
     // If workspace feature is disabled, it will not load the workspace plugin
     if (core.uiSettings.get('workspace:enabled') === false) {
+      // set filtered nav links
```

Comment:
How about move these default nav links setup logic to core plugin? I think these shouldn't be a part of workspace plugin?